test/ijulia.jl

```diff
@@ -0,0 +1,40 @@
+using Conda
+using IJulia
+using RCall
+using Weave
+using Test
+
+# $(abspath(dirname(@__DIR__)))
+IJulia.installkernel("julia", "--project=@.")
+jupyter_path = joinpath(Conda.BINDIR, "jupyter")
+if !isfile(jupyter_path)
+    Conda.add("jupyter")
+end
+testpath = Base.Fix1(joinpath, @__DIR__)
+Weave.notebook(testpath("ijulia.jmd"); out_path=@__DIR__, jupyter_path=jupyter_path)
+
+run(`$(jupyter_path) nbconvert $(testpath("ijulia.ipynb")) --to html --embed-images`)
+const PNG = """<img alt="No description has been provided for this image" class="" src="data:image/png;base64"""
+const SVG = """<img alt="No description has been provided for this image" src="data:image/svg+xml;base64"""
+html = read(testpath("ijulia.html"), String)
+
+# these are the tests to show that things actually work
+@test occursin(PNG,  html)
+@test occursin(SVG,  html)
+
+@test_throws ArgumentError("Unsupported MIME type: lulz") RCall.rdevicename(MIME("lulz"))
+
+# create a folder ijulia_files with the exported images -- could be useful if we ever set up percy
+# run(`$(jupyter_path) nbconvert $(testpath("ijulia.ipynb")) --to markdown`)
+
+# these are the tests to get code coverage
+# nothing like a little piracy when testing in a headless setup
+RCall.ijulia_init()
+R"plot(1:10, 1:10)"
+# throws a method error when running headless !
+@test_throws(MethodError, RCall.ijulia_displayplots())
```

Comment:
@ararslan if you have a better idea, I'd be keen. I considered pirating (in the tests) `display(::MIME, Vector{UInt8})` but that seemed even worse

Reply:
I don't think I fully understand the context. What's the desired/typical behavior? If you want to test that it throws a method error then this seems fine...

Reply:
When called in a Jupyter notebook, this will result in the R plot (written to a temporary file) being read in and then `display`ed using the appropriate MIME type. But when called in a plain text, headless environment, you get a method error. I'm not too worried about it since we're testing the actual behavior separately (via the `nbconvert` stuff above), but not in a way that codecov detects it (due to the nesting of processes). So the run here is just re-running the same code that's in the notebook so that codecov gets it.